src/functions.jl

```diff
@@ -676,16 +699,11 @@ for T in (:Taylor1, :TaylorN)
                 return nothing
             end
             zero!(c, k)
-            if $T == Taylor1
-                @inbounds c[k] = (k-1) * r[1] * c[k-1]
-            else
-                @inbounds mul!(c[k], (k-1) * r[1], c[k-1])
-            end
-            @inbounds for i in 2:k-1
+            @inbounds for i in 1:k-1
                 if $T == Taylor1
                     c[k] += (k-i) * r[i] * c[k-i]
                 else
-                    mul!(c[k], (k-i) * r[i], c[k-i])
+                    mul!(c[k], r[i], c[k-i], scalar=k-i)
                 end
             end
```

Comment:
Same comment about order of `if` and `for`

Reply:
Done 👍